plugins/modules/clusters.py

```diff
@@ -108,9 +113,29 @@ def run_module():
             module.fail_json(msg=f"Error deleting cluster_id: {module.params.get('cluster_id')}", **result)
         
         result = dict(changed=True, clusters=[])
+    # Retrieve cluster
+    elif module.params.get('action') == "get":
 
-    module.exit_json(**result)
+        list_params = {}
+        for k in QUERY_PARAMS_LIST:
+            val = module.params.get(k)
+            if val:
+                list_params = list_params | { k: val }
+
+        cluster_id = module.params.get('cluster_id')
+        request_url = f"{API_URL}/clusters"
+
+        if not module.params.get('cluster_id'):
+            module.fail_json(msg="cluster_id is required for retrieve action")
+        response = requests.get(request_url, params=list_params, headers=headers)
```

Comment:
The only query param in the `/v2/clusters/{cluster_id}` endpoint is `exclude-hosts` and is not defined (yet). If you plan to not support it yet then there is no need to pass params into the request.